docker/user.sh

```diff
@@ -52,6 +52,28 @@ then
     mkdir --parents "$HOME"/output
 fi
 
+# Prepare the python virtual environment. We copy the mounted directory if it exists to `.python_venv` to fix the symbolic
+# links to the Python interpreter in the container without affecting the files on host.
+if [[ -d "$HOME/python_venv" ]];
+then
+    cp -r "$HOME/python_venv" "$HOME/.python_venv"
+fi
+
+if [[ -d "$HOME/.python_venv/bin" ]];
+then
+    python_binaries=(
+        "python"
+        "python3"
+        "python3.11"
+    )
+    for p in "${python_binaries[@]}"; do
+        if [[ -f "$HOME/.venv/bin/${p}" ]];
+        then
+            ln -sf "$HOME/.venv/bin/${p}" "$HOME/.python_venv/bin/${p}"
```

Comment:
I believe it would be useful for adding a comment about why we need to fix the python interpreters within `.python_venv/bin/` (i.e. `cyclone-dx` requires valid python interpreters when it's analyzing a virtual environment).

Reply:
I also wonder why we symlink to the Python interpreters in `$HOME/.venv/bin/${p}` (what we are doing here works fine at the moment, however it doesn't sit right with me).
With virtual environment created by [venv](https://docs.python.org/3/library/venv.html). The Python interpreters are all symlinks to the base Python interpreters that were used to created that environment. 
This will mean that if we creates 2 virtual environments from the Python installation at `/usr/local/bin/python3.11`:
- The Python interpreters of those 2 virtual environments will symlink to `/usr/local/bin/python3.11`
- Both virtual environments should not symlink to each other, because that defeat the idea of isolated virtual environment.
Therefore, I think we should symlink from "$HOME/.python_venv/bin/${p}" to `realpath "$HOME/.venv/bin/python"` (which resolves to `/usr/local/bin/python3.11`, the base Python installation that comes with the base Docker image).